src/lib.rs

```diff
@@ -289,6 +289,9 @@ pub trait OptionExt<T> {
 /// It must be impossible to produce a value of this type.
 pub unsafe trait Never {}
 
+#[inline(always)]
+const fn unreachable_checked<T: Never>(_: T) -> ! { unsafe { unreachable_unchecked() } }
```

Comment:
```suggestion
pub const fn unreachable_checked<T: Never>(_: T) -> ! { unsafe { unreachable_unchecked() } }
```
`Never` for the people! 😄

Reply:
what should the docs say